src/cffsubr/__init__.py

```diff
@@ -197,3 +198,49 @@ def subroutinize(
             post_table.glyphOrder = glyphOrder
 
     return otf
+
+
+def has_subroutines(otf: ttLib.TTFont) -> bool:
+    """Return True if the font's CFF or CFF2 table contains any subroutines."""
+    table_tag = _sniff_cff_table_format(otf)
+    top_dict = otf[table_tag].cff.topDictIndex[0]
+    all_subrs = [top_dict.GlobalSubrs]
+    if hasattr(top_dict, "FDArray"):
+        all_subrs.extend(
+            fd.Private.Subrs for fd in top_dict.FDArray if hasattr(fd.Private, "Subrs")
+        )
+    elif hasattr(top_dict.Private, "Subrs"):
+        all_subrs.append(top_dict.Private.Subrs)
+    return any(all_subrs)
+
+
+def desubroutinize(otf: ttLib.TTFont, inplace=True) -> ttLib.TTFont:
+    """Remove all subroutines from the font.
+
+    Args:
+        otf (ttLib.TTFont): the input font object.
+        inplace (bool): whether to create a copy or modify the input font. By default
+            the input font is modified.
+
+    Returns:
+        The modified font containing the desubroutinized CFF or CFF2 table.
+        This will be a different TTFont object if inplace=False.
+
+    Raises:
+        cffsubr.Error if the font doesn't contain 'CFF ' or 'CFF2' table,
+        or if desubroutinization process fails.
+    """
+    # the 'desubroutinize' method is dynamically added to the CFF table class
+    # as a side-effect of importing the fontTools.subset.cff module...
+    from fontTools.subset import cff as _
+
+    if not inplace:
+        otf = copy.deepcopy(otf)
+
+    table_tag = _sniff_cff_table_format(otf)
+    try:
+        otf[table_tag].desubroutinize()
```

Comment:
`tx` can desubroutinize as well (`tx -cff -S`), no idea if it is any faster.

Reply:
yes, i know https://github.com/adobe-type-tools/cffsubr/pull/5#issuecomment-635936610

I guess it shouldn't matter in the end, as there's only one correct way to desubroutinize. I guess.

Reply:
> if it is any faster

possibly, but desubroutinization is mostly meant for testing/debugging.